packages/medusa/src/services/discount.ts

```diff
@@ -411,7 +407,24 @@ class DiscountService extends BaseService {
       }
 
       if (rule) {
-        discount.rule = this.validateDiscountRule_(ruleToUpdate)
+        let ruleUpdate: Pick<
+          UpdateDiscountRuleInput,
+          "id" | "description" | "value" | "allocation"
+        > = rule
+
+        if (rule.value) {
+          ruleUpdate = this.validateDiscountRule_({
+            ...rule,
+            type: discount.rule.type,
+          })
+        }
+
+        const updatedRule = ruleRepo.create({
+          ...discount.rule,
+          ...ruleUpdate,
+        })
```

Comment:
question: Do we need to create a new entity object here? For updates, we usually follow a different pattern where we, in this case, would do:
```
discount.rule = {
   ...discount.rule,
   ...ruleUpdate
}
```

Reply:
I guess it's not nescessary. But we will then need to type cast the update as as the type of `{
   ...discount.rule,
   ...ruleUpdate
}` !== `DiscountRule`. So it's a matter of what we think is the most clean approach, lmk what you think.

Reply:
Got it - thanks! I'm fine sticking with the current impl. 👍